### ISSUE TITLE
Epic issues Api

### ISSUE THREAD

**Issue by shuklaalok7** [gmessner/gitlab4j-api#215]
This library lacks support for [Epic issues API](https://docs.gitlab.com/ee/api/epic_issues.html). This is available only on `Ultimate` and `Gold` editions of GitLab.
I may add a PR to add this API support. Others' contributions are very much welcome too.

**Comment by gmessner** [comment]
@shuklaalok7 
I am in the process of doing a release with the TagsApi and will add this API.  I just ask that you test it as I do not have access to the Ultimate version of GitLab.

**Comment by shuklaalok7** [comment]
For testing it, one can simply create any open-source project and GitLab enables, I think, `Gold` edition features for that.
I can test it, no problem.

**Comment by gmessner** [comment]
@shuklaalok7 
Are you looking only for the Epic Issues API or both the Epic Issues AOI and the Epics API?

https://docs.gitlab.com/ee/api/epics.html
https://docs.gitlab.com/ee/api/epic_issues.html

While I normally like to have an API class per GitLab API I'm considering combining these two into one class called EpicsApi.  Your thoughts?

**Comment by shuklaalok7** [comment]
I am looking for both these APIs, I was going to create two separate issues for them.

Keeping separate classes for Epics and Epic issues will help in reducing confusion between already confusing named APIs, so I vote for keeping these two classes separate.

**Comment by gmessner** [comment]
I believe having a single API class actually simplifies and clarifies things.  This class (EpicsApi) would include endpoint methods for both the Epics API and Epic Issues API, and this would be very obvious in the Javadocs.  Three classes would be created:

1) EpicsApi - The API class as just described
2) Epic - The model class for objects returned by the GitLab Epics API
3) EpicIssue - The model class for objects returned by the GitLab Epic Issues API (extends the existing Issue class).

I am also working on some additional documentation that basically is a map of the GitLab API doc pages to the GitLab4J API class that implements that specific GitLab API, will make it clear on what GitLab4J API class to use.  Will have this done in a couple weeks.

**Comment by gmessner** [comment]
@shuklaalok7 
I have released 4.8.30 with support for both the Epics API and the Epic Issues API.  I appreciated your input, but decided to go the one API class route. Some of the things that led me that direction was the relatively small size of these, the Epics API web page actually referenced the Epics Issues API, and they both use the same basic endpoints.

I'm going to go ahead and leave this issue open, will allow you to report any bugs found in the implementation.
